### PR TITLE
Optimize docker layers

### PR DESCRIPTION
**Pull request by 412b**
- add classpath jar to artifacts
- move start scripts to the same layer as artifacts, because those contain references to artifacts
- move /jre (JLink) to a separate layer
- move /conf to a separate layer as well, because it's not something changing often

### CODE REVIEW

**Comment by muuki88** [comment]
Thanks for these optimization 🤩 I'll take a deeper look on Monday

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala [hunk 1]
```diff
@@ -105,18 +106,30 @@ object DockerPlugin extends AutoPlugin {
     dockerGroupLayers := {
       val dockerBaseDirectory = (defaultLinuxInstallLocation in Docker).value
       // Ensure this doesn't break even if the JvmPlugin isn't enabled.
-      val artifacts = projectDependencyArtifacts.?.value.getOrElse(Nil).map(_.data).toSet
+      var artifacts = projectDependencyArtifacts.?.value.getOrElse(Nil).map(_.data).toSet
+
+      ClasspathJarPlugin.autoImport.packageJavaClasspathJar.?.value match {
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala:111]: This is probably the easiest way to achieve this :+1: 

I would add a comment why the DockerPlugin cross references another plugin that is unrelated at first sight:

```suggestion
      // add the classpath jar to the project artifacts to improve layer caching as it is created by
      // the ClasspathJarPlugin and is not part of the projectDependencyArtifacts
      ClasspathJarPlugin.autoImport.packageJavaClasspathJar.?.value match {
```

**Review by muuki88 — APPROVED** [review]
This is an amazing improvement. Thanks for the thoughtful pull request :heart:

File: src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala (hunk 1 above)
Comment (412b) [reply src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala:113]: Added that comment :smile:

**Review by muuki88 — APPROVED** [review]
I'll merge and release ASAP :heart:
